src/aics-image-viewer/components/TfEditor/index.tsx

```diff
@@ -356,24 +356,37 @@ export default class MyTfEditor extends React.Component<MyTfEditorProps, MyTfEdi
     svg.selectAll<SVGGElement, unknown>(".axis.axis--x").call(axis);
   }
 
+  private getBinLengths(): { binLengths: number[]; max: number } {
+    const histogram = this.props.channelData.histogram;
+    const binLengths = [];
+    let max = Number.NEGATIVE_INFINITY;
+    for (let i = 0; i < histogram.getNumBins(); i++) {
+      const binLength = histogram.getBin(i);
+      binLengths.push(binLength);
```

Comment:
I wonder if this could be even faster. It looks like it's basically just copying the value of each bin one by one, so could probably do something like histogram.bins.slice() to get binLengths (or even just use it directly without copying if we're sure it won't be modified).  And the max value might already be stored inside the histogram class too. (It ought to be, if it isn't)

Whether it's worth doing this depends on the new timings.  You mentioned 250ms before, but what is the after? (And is this function still the biggest part)

Reply:
I actually couldn't find this function in the performance graph (though admittedly maybe I wasn't looking hard enough)! 

`histogram.bins` is private so I don't have direct access to it unless I perform some unsavory Javascript crimes.

Reply:
ok - we can do some work on volume-viewer to expose that somehow.  In C++ you can expose it as a read-only reference but in js it's all or nothing, I guess.